scripts/remoteperformance.ps1

```diff
@@ -0,0 +1,567 @@
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("Debug", "Release")]
+    [string]$Config = "Debug",
+
+    [Parameter(Mandatory = $false)]
+    [string]$TestsFile = "",
+
+    [Parameter(Mandatory = $false)]
+    [string]$Remote = "",
+
+    [Parameter(Mandatory = $false)]
+    [string]$LocalArch = "x64",
+
+    [Parameter(Mandatory = $false)]
+    [string]$LocalTls = "stub",
+
+    [Parameter(Mandatory = $false)]
+    [string]$RemoteArch = "x64",
+    
+    [Parameter(Mandatory = $false)]
+    [string]$RemoteTls = "stub",
+
+    [Parameter(Mandatory = $false)]
+    [string]$WinRMUser = "",
+
+    [Parameter(Mandatory = $false)]
+    [switch]$SkipDeploy = $false,
+
+    [Parameter(Mandatory = $false)]
+    [switch]$Publish = $false,
+
+    [Parameter(Mandatory = $false)]
+    [switch]$Local = $false
+)
+
+Set-StrictMode -Version 'Latest'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Root directory of the project.
+$RootDir = Split-Path $PSScriptRoot -Parent
+
+Get-PSSession | Remove-PSSession
+
+$LocalPlatform = $null
+if ($IsWindows) {
+    $LocalPlatform = "windows"
+} else {
+    $LocalPlatform = "linux"
+}
+
+
+if ($TestsFile -eq "") {
+    $TestsFile = Join-Path $PSScriptRoot "RemoteTests.json"
+}
+
+function HostToNetworkOrder {
+    param($Address)
+    $Bytes = $Address.GetAddressBytes()
+    [Array]::Reverse($Bytes) | Out-Null
+    return [System.BitConverter]::ToUInt32($Bytes, 0)
+}
+
+if ($Local) {
+    $RemoteAddress = "localhost"
+    $session = $null
+    $LocalAddress = "127.0.0.1"
+} else {
+    if ($Remote -eq "") {
+        if ($WinRMUser -ne "") {
+            $session = New-PSSession -ComputerName quic-server -Credential $WinRMUser -ConfigurationName PowerShell.7
+        } else {
+            $session = New-PSSession -ComputerName quic-server -ConfigurationName PowerShell.7
+        }
+    } else {
+        $session = New-PSSession -HostName "$Remote"
+    }
+
+    $RemoteAddress = $session.ComputerName
+
+    if ($null -eq $session) {
+        exit
+    }
+
+    $PossibleRemoteIPs = [System.Net.Dns]::GetHostAddresses($RemoteAddress) | Select-Object -Property IPAddressToString
+
+    $PossibleLocalIPs = Get-NetIPAddress -AddressFamily IPv4 | Select-Object -Property IPv4Address, PrefixLength
+
+    $MatchedIPs = @()
+
+    $PossibleLocalIPs | ForEach-Object {
+
+        [IPAddress]$LocalIpAddr = $_.IPv4Address
+
+        $ToMaskLocalAddress = HostToNetworkOrder($LocalIpAddr)
+
+        $Mask = (1ul -shl $_.PrefixLength) - 1
+        $Mask = $Mask -shl (32 - $_.PrefixLength)
+        $LocalSubnet = $ToMaskLocalAddress -band $Mask
+
+        $PossibleRemoteIPs | ForEach-Object {
+            [ipaddress]$RemoteIpAddr = $_.IPAddressToString
+            $ToMaskRemoteAddress = HostToNetworkOrder($RemoteIpAddr)
+            $RemoteMasked = $ToMaskRemoteAddress -band $Mask
+
+            if ($RemoteMasked -eq $LocalSubnet) {
+                $MatchedIPs += $LocalIpAddr.IPAddressToString
+            }
+        }
```

Comment:
IMO, this logic is overly complicated. I think we should instead just rely on an ENV var, just like the thumbprint. It would reduce the complexity of this script considerably.

Reply:
I feel like I would rather do it right rather then hardcoding a required env var. I don't like requiring the cert either as it makes testing harder, and after I get done with this I plan on doing some debugging into why PS remoting on those systems specifically doesn't handle self signed certs, as I feel like that might somehow be a bug. With it being a separate function in a separate file now, its much cleaner and easier to maintain.